tests/misc/t20883.nim

```diff
@@ -0,0 +1,12 @@
+discard """
+  action: reject
+  errormsg: "type mismatch: got <float64> but expected 'typeof(U(0.000001))'"
+  line: 8
+  column: 22
+"""
+
+proc foo*[U](x: U = U(1e-6)) =
+  echo x
+
+foo[float]()
+foo()
```

Comment:
This one isn't.

Reply:
yeah, the full stacktrace show the error starting from line 12, as generic proc should Initialization, so the error line is 8